test/type-error.js

```diff
@@ -0,0 +1,105 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/type-error';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const errors = [{
+	ruleId: 'type-error',
+	message: '`new Error()` is too unspecific for a typecheck, use `new TypeError()` instead.'
+}];
+
+ruleTester.run('type-error', rule, {
+	valid: [
+		`if (Array.isArray(foo) || ArrayBuffer.isView(foo)) {
```

Comment:
More test ideas:
- throw new Error(args)
- throw Error()
- throw new Error
- new Error() (not thrown)
- a condition with multiple statements and then a throw
- Throw new Error.foo()
- throw new Foo.Error()
- throw new CustomError()
- with a condition with non-type checking parts

To be clear, not all tests I am requesting will be valid.